Add support for Bareos 22 (rebase)

greatflyingsteve opened the pull request:
#### Pull Request (PR) description
This is a rebase of #129 onto `master` (which should make all the tests pass, now, or at least it does when I run them locally).

#### This Pull Request (PR) fixes the following issues
Fixes #142

aba-rechsteiner committed:
add custom user groups parameter

wolfaba committed:
add repository for windows

wolfaba committed:
do nothing for windows

aba-rechsteiner committed:
add version 22

greatflyingsteve committed:
Update REFERENCE.md

smortex commented on manifests/init.pp:
```diff
@@ -38,7 +38,8 @@
   $file_mode                          = $bareos::params::file_mode,
   $file_dir_mode                      = $bareos::params::file_dir_mode,
-  String  $repo_release               = '21',
+  $user_groups                        = $bareos::params::user_groups,
```
Can you add a data type for this new parameter?  Probably `Array[String[1]]`.

greatflyingsteve replied:
If you don't mind, I'd like to add that as part of attempting to get #154 incorporated, as well.  If you do mind, of course, I'm not opposed to adding it, here, it just feels vaguely like making a style shift like that all at once is "cleaner?"

greatflyingsteve replied:
> Also, as it is a new parameter, do not put the default value in `params.pp` but put it here. That way, the actual value is documented when you generate the documentation.

I was mostly aiming to port the original author's work mostly-unmolested, but I also find the use of params.pp here icky.  Is it better to try and stick with a clean rebase, or should I just go ahead and fix things While I'm In Here™?

smortex commented on manifests/init.pp:
```diff
@@ -38,7 +38,8 @@
   $file_dir_mode                      = $bareos::params::file_dir_mode,
-  String  $repo_release               = '21',
+  $user_groups                        = $bareos::params::user_groups,
+  String  $repo_release               = '22',
```
Changing the default version make it a breaking change. This is just a FYI, if the latest version is 22 we can change the default in the next (major) version. I will add a backward-incompatible label accordingly.

smortex replied:
I see #161 is about adding support for bareos 23. Do you plan to rebase this version too next?

greatflyingsteve committed:
Style updates: avoid params.pp, add typing

Freshen code style to bring things more up to date and avoid legacy
patterns like params.pp and untyped parameters.  Includes updates to
REFERENCE.md.
